### ISSUE TITLE
Muscles that are disabled still produce activations in Static Optimization

### ISSUE THREAD

**Issue by jimmyDunne** [opensim-org/opensim-core#2438]
If you have muscles enabled == 0, activations are still produced for the disabled muscles that are colored and are also written to file. Below, the TRIlat and TRImed are disabled during the playback of a Static Optimization analysis. 

![arm25_SO](https://user-images.githubusercontent.com/5296507/56435006-e5bc6b00-62a4-11e9-8644-207e1932e5c9.gif)

Once the simulation is complete, the output activation files contain information for the muscles that should not have been part of the simulation. Below the TRIlat & TRImed (Green) produce activation.

![image](https://user-images.githubusercontent.com/5296507/56434728-d1c43980-62a3-11e9-9bca-bcd3612eb3b1.png)

There is no force information for the TRIlat and TRImed muscles written to file. Having any activations generated for disabled muscles is confusing, both in the visualization and in the output files. This will be fixed when disabled muscles do not produce any activations. 

A package with files need to replicate the issue is posted below;
[disabled_Muscle_SO.zip](https://github.com/opensim-org/opensim-core/files/3098811/disabled_Muscle_SO.zip)

**Comment by jenhicks** [comment]
Thanks for reporting @jimmyDunne . I was able to reproduce this issue. Two comments/questions:

- I believe this is an issue only with Static Optimization. Forward and CMC seem to respect when muscles are disabled. (I ran CMC to track the same motion and the disabled muscles were not activated. I ran a forward simulation using the SO controls and the disabled muscles were not activated.)

- There is a second potential GUI issue. @aymanhab : disabled muscles seem to show up as always red when you visualize a states file, which is confusing (since red usually means activated). Is there already an issue for this or have we already discussed previously?

**Comment by aymanhab** [comment]
@jenhicks As far as I know, the muscles provide the color (unfortunate in my opinion but true), when playing back motion or during simulation this is based on the state. The GUI does not manufacture colors on its own. We can change the behavior of the muscles so that they show some shade of gray if not activated, but I wouldn't call this strictly a GUI issue. I was advocating earlier in 4.0 that we shouldn't use any colors in the red-blue spectrum as default color to avoid confusion but we collectively decided not to change how models looked from 3.3 to 4.0. I don't believe there's an issue so feel free to open a new one.

**Comment by aymanhab** [comment]
Indeed the files contain activations for disabled muscles which are used downstream to color. The GUI colors by values from a file regardless of the source of this file. Now looking into SO itself which seems to be the offending party.

**Comment by aymanhab** [comment]
It appears SO never considered the enabled/appliesForce flag at all. All loops cycle through actuators without ever checking the flag AFAIK,  Please let me know if you have reason to believe otherwise @jimmyDunne @tkuchida @aseth1 Not clear how involved this fix will be.

**Comment by tkuchida** [comment]
It doesn't look like [the SO test](https://github.com/opensim-org/opensim-core/blob/master/Applications/Analyze/test/testStaticOptimization.cpp) covers this scenario. There should be a test case where a muscle is set to enable=false and throws if the muscle force is nonzero. #1188

**Comment by aymanhab** [comment]
PR in place to fix the issue and produce result files with no activations or forces for disabled actuators. Color of disabled muscles stay fixed during playback but now is red (because that's what's in the state) 😕 Investigating if we can change that to something else (either :blue or gray)

**Comment by aymanhab** [comment]
I'm working on a fix to the coloring issue in the GUI but I think this issue is separate from wrong SO results. I can open a separate issue and we close this one.  Thoughts @jimmyDunne @tkuchida

**Comment by tkuchida** [comment]
@aymanhab I agree with opening a new issue over at [opensim-gui](https://github.com/opensim-org/opensim-gui/issues).

**Comment by aymanhab** [comment]
This can be closed, GUI issue https://github.com/opensim-org/opensim-gui/issues/1123  opened.
